### PR TITLE
Fixed error handling for bookmarks manager

### PR DESCRIPTION
**Pull request by jjfreund**

### CODE REVIEW

File: internal/bookmarks/manager.go [hunk 1]
```diff
@@ -30,10 +30,24 @@ func (m *Manager) Get(w http.ResponseWriter, r *http.Request) {
 	userId := r.URL.Query().Get("user_id")
 
 	if userId != "" {
-		iUserId, _ := strconv.Atoi(userId)
-		dbBookmarks = m.DbClient.GetBookmarksByUserID(iUserId)
+		iUserId, err := strconv.Atoi(userId)
+		if (err != nil) {
```
Comment (lgarofalo) [inline internal/bookmarks/manager.go:34]: dont need to wrap this, normal to just do
```
if err != nil {
 stuff;
}
```

File: internal/bookmarks/manager.go [hunk 2]
```diff
@@ -30,10 +30,24 @@ func (m *Manager) Get(w http.ResponseWriter, r *http.Request) {
...
+			log.Printf("%v", err)
+			writeStatus(w, http.StatusBadRequest)
```
Comment (lgarofalo) [inline internal/bookmarks/manager.go:36]: This is a case where we can update the flow to provide an error message json body response

File: internal/bookmarks/manager.go [hunk 3]
```diff
@@ -46,9 +60,15 @@ func (m *Manager) GetByID(w http.ResponseWriter, r *http.Request) {
 	id, err := strconv.Atoi(chi.URLParam(r, "id"))
 	if err != nil {
 		log.Printf("%v", err)
+		writeStatus(w, http.StatusBadRequest)
+		return		
 	}
 
-	dbBookmark := m.DbClient.GetBookmarkByID(id)
+	dbBookmark, err := m.DbClient.GetBookmarkByID(id)
+	if (err != nil) {
```
Comment (lgarofalo) [inline internal/bookmarks/manager.go:68]: We want to log the error here too, just the same flow as other cases

File: internal/bookmarks/manager.go (hunk 2 above)
Comment (lgarofalo) [inline internal/bookmarks/manager.go:36]: I can add a test for any 400 cases

File: internal/db/manager.go
Comment (lgarofalo) [inline internal/db/manager.go:1]: makes sense, we do want to bubble up errors, it was a mistake of me to not do that originally

**Comment by lgarofalo** [comment]
Updated PR with responses to my comments

**Review by richardxia — APPROVED** [review]
This LGTM, and thanks, @lgarofalo, for pushing up a commit to create a common Error struct for returning error responses with JSON bodies!

**Pull request merged**